packages/core/src/physics/Collider.ts

```diff
@@ -0,0 +1,100 @@
+import { Component } from "../Component";
+import { ignoreClone } from "../clone/CloneManager";
+import { ICollider } from "@oasis-engine/design";
+import { ColliderShape } from "./shape/ColliderShape";
+import { UpdateFlag } from "../UpdateFlag";
+import { Entity } from "../Entity";
+
+/**
+ * Abstract class for collider shapes.
+ */
+export abstract class Collider extends Component {
+  /** @internal */
+  @ignoreClone
+  _index: number = -1;
+  /** @internal */
+  _nativeCollider: ICollider;
+
+  protected _updateFlag: UpdateFlag;
+
+  private _shapes: ColliderShape[] = [];
+
+  /**
+   * The shapes of this collider.
+   */
+  get shapes(): Readonly<ColliderShape[]> {
+    return this._shapes;
+  }
+
+  protected constructor(entity: Entity) {
+    super(entity);
+    this._updateFlag = this.entity.transform.registerWorldChangeFlag();
+  }
+
+  /**
+   * Add collider shape on this collider.
+   * @param shape collider shape.
+   */
+  addShape(shape: ColliderShape): void {
+    shape._collider = this;
+    this._nativeCollider.addShape(shape._nativeShape);
+    this._shapes.push(shape);
+  }
+
+  /**
+   * Remove a collider shape.
+   * @param shape - The collider shape.
+   */
+  removeShape(shape: ColliderShape): void {
+    this._nativeCollider.removeShape(shape._nativeShape);
+    const index = this._shapes.indexOf(shape);
+    index != -1 && this._shapes.splice(index, 1);
+  }
+
+  /**
+   * Remove all shape attached.
+   */
+  clearAllShape(): void {
+    let shapes = this._shapes;
+    for (let i = 0, len = shapes.length; i < len; i++) {
+      this.removeShape(shapes[i]);
+    }
+  }
+
+  /**
+   * @internal
+   */
+  _onStart() {
```

Comment:
'_onStart' mean only call once, for example `Script`'s `onStart()` call back. You can name it `_onUpdate`

Reply:
done